org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsi/participants/XSIFormatterParticipant.java

```diff
@@ -197,11 +200,20 @@ public boolean formatAttributeValue(DOMAttr attr, XMLFormatterDocumentNew format
 		int lineFeed = split == XSISchemaLocationSplit.onElement ? 1 : 2;
 		int locationNum = 1;
 		String attrValue = attr.getOriginalValue();
+		int lastAttrValueTermIndex = 0;
+		int availableLineWidth = parentConstraints.getAvailableLineWidth();
 
 		for (int i = firstContentOffset; i < attrValue.length(); i++) {
 			int from = formatterDocument.adjustOffsetWithLeftWhitespaces(attrValueStart, attrValueStart + i + 1);
 			if (Character.isWhitespace(attrValue.charAt(i)) && !Character.isWhitespace(attrValue.charAt(i + 1))
 					&& !StringUtils.isQuote(attrValue.charAt(from - attrValueStart))) {
+				availableLineWidth -= i - lastAttrValueTermIndex;
+				lastAttrValueTermIndex = i;
+				if (availableLineWidth < 0 && formatterDocument.getMaxLineWidth() != 0

```

Comment:
Instead of writing formatterDocument.getMaxLineWidth() != 0 I will introduce a new method  like

formatterDocument.isMaxLineWidthSupported()

It will give the capability to differentiate the use of value of maxLineWith and the check if maxLineWith is enabled.